Update CITATION.cff with contributors for 0.5.0

jni opened the pull request:
Thanks to @Czaki for reminding me that there's no need to do this manually when he created a [script](https://github.com/napari/napari-release-tools/blob/main/find_contributors_without_citation.py) to get this info. 😅

**Edit:** categorising into not responded/added/declined:

Not yet responded:
@AmirAflak @DanGonite57 @mstabrin 

Added:
@AndrewAnnex @MarchisLost @aeisenbarth @imagejan @jamesyan-git @jo-mueller @maweigert @odinsbane @palec87 @perlman

Declined:
@stefanv

Thank you for contributing to napari! If you would like to be included in the CITATION.cff file for the next release, please comment by filling in the following template (in a code block) and I'll add your response to the file:

```
- given-names:
  family-names:
  affiliation:
  orcid: (optional, format as url: https://orcid.org/0000-0000-0000-0000)
  alias: (gh handle)
```

jni committed:
Empty commit to start PR to update CITATION.cff

jni committed:
Add Johannes

Czaki commented on CITATION.cff:
```diff
@@ -255,6 +255,11 @@ authors:
+- given-names: Johannes
+  family-names: Soltwedel
+  affiliation: DFG cluster of excellence 'Physics of Life', TU Dresden
+  orcid: 0000-0003-1273-2412
```
```suggestion
  orcid: https://orcid.org/0000-0003-1273-2412
```

Validation workflow is working :)

jni replied:
Nice! So annoying though that it needs the URL. I'll update my template above.

Czaki replied:
Lack of URL results in failure to deploy of release to zenodo.

jni replied:
Yes, I remember now 😅, I'm just saying, that seems like a failure in Zenodo/the cff format.

jni committed:
Include orcid as URL

Co-authored-by: Grzegorz Bokota <bokota+github@gmail.com>

jni committed:
Add responders:

Andrew Annex
Jan Eglinger
Andreas Eisenbarth
David Palecek
Eric Perlman
David Pinto

jni committed:
Add James

jni committed:
Add Martin

Czaki commented on CITATION.cff:
```diff
@@ -227,6 +252,9 @@ authors:
   alias: kir0ul
+- given-names: David
+  family-names: Pinto
+  alias: MarchIsLost
```
```suggestion
  alias: MarchisLost
```

jni committed:
Add MB

jni committed:
Fix capitalisation on username

Co-authored-by: Grzegorz Bokota <bokota+github@gmail.com>

jni committed:
Merge branch 'main' into missing-persons
